v0/destinations/signl4/transform.js

```diff
@@ -0,0 +1,114 @@
+const { EventType } = require("../../../constants");
+const { BASE_URL } = require("./config");
+const { populatePayload } = require("./utils");
+const {
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  CustomError,
+  getErrorRespEvents,
+  getSuccessRespEvents,
+  removeUndefinedAndNullAndEmptyValues
+} = require("../../util");
+
+const responseBuilder = (payload, endpoint) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = endpoint;
+    response.headers = {
+      "Content-Type": "application/json"
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullAndEmptyValues(payload);
+    return response;
+  }
+  throw new CustomError(
+    "Monday]: Payload could not be populated due to wrong input",
+    400
+  );
+};
+
+/**
+ * This function is used to build the response for track call.
+ * @param {*} message
+ * @param {*} param1
+ * @returns
+ */
+const trackResponseBuilder = async (message, { Config }) => {
+  const { apiKey } = Config;
+  const { event } = message;
+
+  if (!event) {
+    throw new CustomError(
+      "[SIGNL4]: event is not present in the input payload",
+      400
+    );
+  }
+
+  const endpoint = `${BASE_URL}/${apiKey}`;
+
+  const payload = populatePayload(message, Config);
+
+  return responseBuilder(payload, endpoint);
+};
+
+const processEvent = async (message, destination) => {
+  if (!message.type) {
+    throw new CustomError(
+      "Message Type is not present. Aborting message.",
+      400
+    );
+  }
+  if (!destination.Config.apiKey) {
+    throw new CustomError("[SIGNL4]: apiKey is a required field", 400);
+  }
+  const messageType = message.type.toLowerCase();
+  let response;
+  switch (messageType) {
+    case EventType.TRACK:
+      response = await trackResponseBuilder(message, destination);
```

Comment:
Why are you using await here? We are not making any intermediate Axios calls in `trackResponseBuilder`.

Reply:
Removed